llm_on_ray/finetune/finetune.py

```diff
@@ -190,6 +110,75 @@ def convert_dtype(dtype: str) -> Optional[torch.dtype]:
     return supported_dtypes[dtype]
 
 
+def load_tokenizer(config: Dict):
+    name = config.get("name")
+    load_config = config.get("config", {})
+    tokenizer = transformers.AutoTokenizer.from_pretrained(name, **load_config)
+    return tokenizer
+
+
+def load_datasets(config: Dict):
+    def local_load(name, **load_config):
+        if os.path.isfile(name):
+            file = os.path.basename(os.path.abspath(name))
+            path = os.path.dirname(os.path.abspath(name))
+            dataset = datasets.load_dataset(path, data_files=file, **load_config)
+        else:
+            dataset = datasets.load_dataset(name, **load_config)
+        return dataset["train"]
+
+    name = config.get("name")
+    load_from_disk = config.get("load_from_disk", False)
+    validation_file = config.get("validation_file", None)
+    validation_split_percentage = config.get("validation_split_percentage", 0)
+
+    if name is not None and os.path.exists(name):
+        train_dataset = local_load(name)
+        if validation_file is not None:
+            validation_dataset = local_load(validation_file)
+            return datasets.DatasetDict({"train": train_dataset, "validation": validation_dataset})
+        if validation_split_percentage / 100 > 0.0 and validation_split_percentage / 100 < 1.0:
+            datasets_dict = train_dataset.train_test_split(
+                test_size=validation_split_percentage / 100
+            )
+            datasets_dict["validation"] = datasets_dict["test"]
+            return datasets_dict
+        return datasets.DatasetDict({"train": train_dataset})
+    else:
+        load_config = config.get("load_config", {})
+        if load_from_disk:
+            return datasets.load_from_disk(name, **load_config)
+        else:
+            return datasets.load_dataset(name, **load_config)
```

Comment:
In my opinion, the validation_split_percentage parameter from the Hugging Face dataset library is a useful feature for splitting the validation dataset. During my development of DPO, I utilized this parameter to split the validation data from the original dataset. This method ensures a straightforward and consistent way to create validation datasets by specifying the percentage of data to be used for validation.
        if data_args.dataset_name is not None:
            # Downloading and loading a dataset from the hub.
            raw_datasets = load_dataset(
                data_args.dataset_name,
                data_args.dataset_config_name,
                cache_dir=model_args.cache_dir,
                token=model_args.token,
                streaming=data_args.streaming,
            )

            if "validation" not in raw_datasets.keys() and training_args.do_eval:
                raw_datasets["validation"] = load_dataset(
                    data_args.dataset_name,
                    data_args.dataset_config_name,
                    split=f"train[:{data_args.validation_split_percentage}%]",
                    cache_dir=model_args.cache_dir,
                    token=model_args.token,
                    streaming=data_args.streaming,
                )
                raw_datasets["train"] = load_dataset(
                    data_args.dataset_name,
                    data_args.dataset_config_name,
                    split=f"train[{data_args.validation_split_percentage}%:]",
                    cache_dir=model_args.cache_dir,
                    token=model_args.token,
                    streaming=data_args.streaming,
                )